### PR TITLE
Fix nav item vertical group spacing

### LINKED ISSUE #3101: The examples for vertical group in navigation item have incorrect spacing
### Package name(s)

Website / Documentation

### Package version(s)

_No response_

### Description

The vertical group examples ([1](https://www.saltdesignsystem.com/salt/components/navigation-item/examples#vertical-group), [2](https://www.saltdesignsystem.com/salt/components/navigation-item/examples#vertical-nested-group)) on the navigation item documentation page have incorrect spacing according to the [vertical navigation pattern](https://www.saltdesignsystem.com/salt/patterns/vertical-navigation).

### PR DESCRIPTION
**Pull request by libertymayc**
closes #3101

### CODE REVIEW

File: site/src/examples/navigation-item/VerticalNestedGroup.tsx [hunk 1]
```diff
@@ -23,7 +23,15 @@ export const VerticalNestedGroup = (): ReactElement => {
 
   return (
     <nav>
-      <ul style={{ width: 250, listStyle: "none", paddingLeft: 0 }}>
+      <StackLayout
```
Comment (joshwooding) [inline site/src/examples/navigation-item/VerticalNestedGroup.tsx:26]: StackLayout isn't really giving us a lot of benefits here over just:

display: flex;
flex-drection: column

**Review by joshwooding — CHANGES_REQUESTED** [review]
See comment

File: site/src/examples/navigation-item/VerticalNestedGroup.tsx (hunk 1 above)
Comment (libertymayc) [reply site/src/examples/navigation-item/VerticalNestedGroup.tsx:26]: gotcha - but the pattern also says [https://www.saltdesignsystem.com/salt/patterns/vertical-navigation#layout ](Vertical navigation uses StackLayout to display navigation items in a column formation.)

**Comment by lukeac123** [comment]
Whats the plan with this, we gonna change the pattern docs or is this good to commit ? @joshwooding @libertymayc

**Comment by libertymayc** [comment]
> Whats the plan with this, we gonna change the pattern docs or is this good to commit ? @joshwooding @libertymayc

unsure, @joshwooding ?

**Comment by joshwooding** [comment]
> > Whats the plan with this, we gonna change the pattern docs or is this good to commit ? @joshwooding @libertymayc
> 
> 
> 
> unsure, @joshwooding ?

We're going to wait for #3242

**Review by joshwooding — APPROVED** [review]

**Review by Fercas123 — APPROVED** [review]

**Pull request merged**